tests/validations/mta/mtaDiagnostic.spec.ts

```diff
@@ -0,0 +1,165 @@
+import { Disposable, mockVscode, testVscode } from "../../mockUtil";
+mockVscode("src/validations/mta/mtaDiagnostic");
+import * as sinon from "sinon";
+import { expect } from "chai";
+import { mta } from "@sap/mta-lib";
+import {
+  clearCurrentDiagnosticCollections,
+  clearDiagnosticCollections,
+  convertMtaIssueCoordinateToEditorCoordinate,
+  diagnosticCollections,
+  getDiagnosticsCollection,
+  mtaIssueToEditorCoordinate,
+} from "../../../src/validations/mta/mtaDiagnostic";
+import { keys } from "lodash";
+
+describe("mtaDiagnostic", () => {
+  const disposables: Disposable[] = [];
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe("diagnosticCollections tests", async () => {
+    beforeEach(() => {
+      clearCurrentDiagnosticCollections();
+    });
+
+    it("getDiagnosticsCollection returns cached DiagnosticCollection", async () => {
+      const diagnosticsName = "diagnosticsName";
+      const diagnosticCollection = {};
+      const createDiagnosticCollectionSpy = sinon
+        .stub(testVscode.languages, "createDiagnosticCollection")
+        .returns(diagnosticCollection);
```

Comment:
Instead of `returns` maybe this should do `callsFake` so a new object is returned every time. You don't really have to check later on that the returned value is what you return here, you can just check the 2 returned diagnostic collections are equal to each other.

Reply:
cache means that you get the same instance. not the same value. 
I can change it if you want bug I think this is better

Reply:
`expect(A).to.equal(B)`, when A and B are objects, will check it's the same instance
https://www.chaijs.com/api/bdd/#method_equal
> Asserts that the target is strictly (===) equal to the given val.

Reply:
You want to check that the cache is implemented in the productive function, it will be easier if you return a new object every time (then the equality assertions will fail if the cache is not implemented, which is not the case now)